src/services/ApiKeyService.ts

```
@@ -38,6 +52,36 @@ export function notifyApiKeyUnset() {
   }
 }
 
+/**
+ * Save API keys to the backend
+ */
+export async function saveApiKeysToBackend(apiKeys: IntegrationTokensPatchRequest) {
+  console.log('ApiKeyService: saveApiKeysToBackend called with:', apiKeys);
+
+  return osidbFetch({
+    method: 'PATCH',
+    url: '/osidb/integrations',
+    data: apiKeys,
+  })
+    .catch(createCatchHandler('Failed to save API keys'));
+}
+
+/**
+ * Retrieve API keys from the backend
+ */
+export async function getApiKeysFromBackend(): Promise<IntegrationTokensResponse> {
+  try {
+    const response = await osidbFetch({
+      method: 'get',
+      url: '/osidb/integrations',
+    });
+    return response.data as IntegrationTokensResponse;
+  } catch (error) {
+    console.error('Failed to retrieve API keys:', error);
+    throw error;
```

Do we want to re-throw the error or return a value that indicates a fetching failure here?

---

We don't have a solid pattern for error handling, so I opted for re-throwing the error just cause some operations   legitimately need to fail if they can't get API keys, but not strong preference.